src/d3/scatterplot.d3.ts

```diff
@@ -9,75 +9,101 @@
 
 // for running on unit tests
 import * as d3 from 'd3';
+import * as AFRAME from 'aframe';
 import * as THREE from 'three';
+import { TimeSeriesPoint } from '../datasets/queries/time-series.query';
+import { VRTimeSeriesPoint } from '../datasets/queries/vr-time-series.query';
 
 export class Scatterplot{
-    private data: number[];
+    private data: TimeSeriesPoint[] | VRTimeSeriesPoint[];
     private shape: string;
     private container: HTMLElement | null;
-    private hscale: d3.ScaleLinear<number, number>;
+    private dataType: string;
+    private timeScale: d3.ScaleTime<number, number>;
+
 
   constructor(shape: string) {
     this.shape = shape;
   }
-  init(container: HTMLElement, data: number[]){
+  init(container: HTMLElement, data: TimeSeriesPoint[] | VRTimeSeriesPoint[], dataType: string){
+    console.log(data);
     this.data = data;
+    this.dataType = dataType;
     this.container = container;
-    document.body.append(container);
+    document.body.append(this.container);
     // create a scale so that there is correspondence between data set and screen render
-    this.hscale = d3.scaleLinear();
-    this.hscale.domain([0, d3.max(this.data) as number])       // max of dataset
-    .range([0, 100]);                                      // linear mapping of data set values to values from 0 to 10
-    this.createSky();
-    this.generatePts();
-    this.setColor('blue');
+   this.generatePts();
+   this.setColor('blue');
+   this.createSky('gray');
     this.createGridPlane();
   }
+  private scaleTime(date: Date): number{
+    console.log(this.data[0].x);
+    console.log(this.data[this.data.length -1].x);
+    const startTime = this.data[0].x;
+    const endTime = this.data[this.data.length -1].x;
+    this.timeScale = d3.scaleTime().domain([startTime, endTime]).rangeRound([0, 31]);
```

Comment:
`31`, probably make this an input argument or a constant.

Reply:
setting 31 as a constant called `TIME_MAX`